### PR TITLE
api: improve documentation of ContainerConfig type

### PR DESCRIPTION
**Pull request by thaJeztah**
- addresses https://github.com/docker/cli/issues/2868
- relates to https://github.com/moby/moby/issues/38762
- relates to https://github.com/moby/moby/issues/17780

ContainerConfig is used in multiple locations (for example, both for
Image.Config and Image.ContainerConfig). Unfortunately, swagger does
not allow documenting individual uses if a type is used; for this type,
the content is _optional_ when used as Image.ContainerConfig (which is
set by the classic builder, which does a "commit" of a container, but
not used when building an image with BuildKit).

This patch attempts to address this confusion by documenting that
"it may be empty (or fields not propagated) if it's used for the
Image.ContainerConfig field".

Perhaps alternatives are possible (aliasing the type?) but we can
look at those in a follow-up.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
@tonistiigi PTAL

File: api/types/types.go [hunk 1]
```diff
@@ -75,6 +75,10 @@ type ImageInspect struct {
 
 	// ContainerConfig is the configuration of the container that was committed
 	// into the image.
+	//
+	// Depending on how the image was created, ContainerConfig, or its fields
```
Comment (tonistiigi) [inline api/types/types.go:79]: For the `ContainerConfig` field inside image I would say:

ContainerConfig is an optional field containing the configuration of the container that was last committed when creating the image. Previous versions of Docker builder used this field to store build cache and it is not in active use anymore.

**Comment by thaJeztah** [comment]
@tonistiigi updated; PTAL

**Review by tonistiigi — APPROVED** [review]

**Comment by thaJeztah** [comment]
Thx! Let me bring this one in

**Pull request merged**